extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -177,121 +190,122 @@ private void restartJobs() {
-                            jobDetail = newJob(getDailyJobClass()).withIdentity("job-daily-startup-" + typeId, thingUid)
-                                    .usingJobData(jobDataMap).build();
-                            quartzScheduler.scheduleJob(jobDetail, trigger);
-
-                            if (isPositionalChannelLinked()) {
-                                // positional intervalJob
-                                jobDataMap = new JobDataMap();
-                                jobDataMap.put(AbstractBaseJob.KEY_THING_UID, thingUid);
-                                jobDataMap.put(AbstractBaseJob.KEY_JOB_NAME, "job-positional");
-
-                                Date start = new Date(System.currentTimeMillis() + (thingConfig.getInterval()) * 1000);
-                                trigger = newTrigger().withIdentity("trigger-positional", thingUid).startAt(start)
-                                        .withSchedule(simpleSchedule().repeatForever()
-                                                .withIntervalInSeconds(thingConfig.getInterval()))
-                                        .build();
-                                jobDetail = newJob(PositionalJob.class).withIdentity("job-positional", thingUid)
-                                        .usingJobData(jobDataMap).build();
-                                quartzScheduler.scheduleJob(jobDetail, trigger);
-                                logger.info("Scheduled astro job-positional with interval of {} seconds for thing {}",
-                                        thingConfig.getInterval(), thingUid);
-                            }
-                        }
+                monitor.lock();
+                if (getThing().getStatus() == ONLINE) {
+                    final String thingUID = getThing().getUID().toString();
+                    final String typeId = getThing().getThingTypeUID().getId();
+                    // Daily Job
+                    final Optional<Job> dailyJob = getDailyJob();
+                    if (!dailyJob.isPresent()) {
+                        logger.error("Daily Job instance not present");
+                        return;
+                    }
+                    if (!scheduledExecutor.isPresent()) {
+                        logger.warn("Thread Pool Executor is not available");
+                        return;
+                    }
+                    final ExpressionThreadPoolExecutor executor = scheduledExecutor.get();
+                    final Expression midNightExpression = new CronExpression(DAILY_MIDNIGHT);
+                    executor.schedule(dailyJob.get(), midNightExpression);
+                    logger.info("Scheduled astro job-daily-{} at midnight for thing {}", typeId, thingUID);
+
+                    // Startup Job
+                    final Calendar now = Calendar.getInstance();
+                    final Expression dateExpression = new CronExpression(createCronFromDate(now.getTime()));
```

Comment:
It is a cron expression and the local variable name has been chosen `dateExpression` as it gets created from Date. That's the reason. For better comprehension, I will change it to a suitable name too.